observatory.go

```diff
@@ -0,0 +1,403 @@
+	URL    string
+	Region string
+}
+
+type ObservatoryPageTestResponse struct {
+	Response
+	Result ObservatoryPageTest `json:"result"`
+}
+
+// CreateObservatoryPageTest starts a test for a page in a specific region.
+//
+// API reference: https://api.cloudflare.com/#speed-create-test
+func (api *API) CreateObservatoryPageTest(ctx context.Context, rc *ResourceContainer, params CreateObservatoryPageTestParams) (*ObservatoryPageTest, error) {
+	if params.URL == "" {
+		return nil, ErrMissingObservatoryUrl
+	}
+	uri := fmt.Sprintf("/zones/%s/speed_api/pages/%s/tests", rc.Identifier, url.PathEscape(params.URL))
+	res, err := api.makeRequestContext(ctx, http.MethodPost, uri, struct {
+		Region string `json:"region"`
+	}{
+		Region: params.Region,
+	})
+	if err != nil {
+		return nil, err
+	}
+	var r ObservatoryPageTestResponse
+	err = json.Unmarshal(res, &r)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+	return &r.Result, nil
+}
+
+type DeleteObservatoryPageTestsParams struct {
+	URL    string
+	Region string `url:"region"`
+}
+
+type ObservatoryCountResponse struct {
+	Response
+	Result struct {
+		Count int `json:"count"`
+	} `json:"result"`
+}
+
+// DeleteObservatoryPageTests deletes all tests for a page in a specific region.
+//
+// API reference: https://api.cloudflare.com/#speed-delete-tests
+func (api *API) DeleteObservatoryPageTests(ctx context.Context, rc *ResourceContainer, params DeleteObservatoryPageTestsParams) (*int, error) {
+	if params.URL == "" {
+		return nil, ErrMissingObservatoryUrl
+	}
+	uri := buildURI(fmt.Sprintf("/zones/%s/speed_api/pages/%s/tests", rc.Identifier, url.PathEscape(params.URL)), params)
+	res, err := api.makeRequestContext(ctx, http.MethodDelete, uri, nil)
+	if err != nil {
+		return nil, err
+	}
+	var r ObservatoryCountResponse
+	err = json.Unmarshal(res, &r)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+	return &r.Result.Count, nil
+}
+
+type GetObservatoryPageTestParams struct {
+	URL    string
+	TestID string
+}
+
+// GetObservatoryPageTest returns a specific test for a page.
+//
+// API reference: https://api.cloudflare.com/#speed-get-test
+func (api *API) GetObservatoryPageTest(ctx context.Context, rc *ResourceContainer, params GetObservatoryPageTestParams) (*ObservatoryPageTest, error) {
+	if params.URL == "" {
+		return nil, ErrMissingObservatoryUrl
+	}
+	if params.URL == "" {
+		return nil, ErrMissingObservatoryTestID
+	}
+	uri := fmt.Sprintf("/zones/%s/speed_api/pages/%s/tests/%s", rc.Identifier, url.PathEscape(params.URL), params.TestID)
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return nil, err
+	}
+	var r ObservatoryPageTestResponse
+	err = json.Unmarshal(res, &r)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+	return &r.Result, nil
+}
+
+type CreateObservatoryScheduledPageTestParams struct {
+	URL       string
```

Comment:
i think you'll want this to prevent it from ending up in the URL

```suggestion
	URL       string `url:"-"`
```